In [3]:
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

In [2]:
import pandas as pd 
import numpy as np

In [4]:
cases = pd.read_csv('../data/cases.csv')
tran = pd.read_csv('../data/tran.csv')
country = pd.read_csv('../data/country.csv')
cluster = pd.read_csv('../data/cluster.csv')

In [5]:
cases.head()

Case no. Report date Date of onset Gender  Age  Name of hospital admitted  \
0         1  23/01/2020    21/01/2020      M   39                        NaN   
1         2  23/01/2020    18/01/2020      M   56                        NaN   
2         3  24/01/2020    20/01/2020      F   62                        NaN   
3         4  24/01/2020    23/01/2020      F   62                        NaN   
4         5  24/01/2020    23/01/2020      M   63                        NaN   

  Hospitalised/Discharged/Deceased HK/Non-HK resident Case classification*  \
0                       Discharged    Non-HK resident        Imported case   
1                       Discharged        HK resident        Imported case   
2                       Discharged    Non-HK resident        Imported case   
3                       Discharged    Non-HK resident        Imported case   
4                       Discharged    Non-HK resident        Imported case   

  Confirmed/probable  
0          Confirmed  
1          Confirmed  
2          Confirmed  
3          Confirmed  
4          Confirmed

In [7]:
cases.shape

(5702, 10)

In [8]:
cases.index

RangeIndex(start=0, stop=5702, step=1)

In [9]:
cases.columns

Index(['Case no.', 'Report date', 'Date of onset', 'Gender', 'Age',
       'Name of hospital admitted', 'Hospitalised/Discharged/Deceased',
       'HK/Non-HK resident', 'Case classification*', 'Confirmed/probable'],
      dtype='object')

In [10]:
cases.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5702 entries, 0 to 5701
Data columns (total 10 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   Case no.                          5702 non-null   int64  
 1   Report date                       5702 non-null   object 
 2   Date of onset                     5702 non-null   object 
 3   Gender                            5702 non-null   object 
 4   Age                               5702 non-null   int64  
 5   Name of hospital admitted         0 non-null      float64
 6   Hospitalised/Discharged/Deceased  5702 non-null   object 
 7   HK/Non-HK resident                5702 non-null   object 
 8   Case classification*              5702 non-null   object 
 9   Confirmed/probable                5702 non-null   object 
dtypes: float64(1), int64(2), object(7)
memory usage: 445.6+ KB


In [11]:
cases.describe()

Case no.          Age  Name of hospital admitted
count  5702.000000  5702.000000                        0.0
mean   2851.500000    43.944756                        NaN
std    1646.169949    20.005356                        NaN
min       1.000000     0.000000                        NaN
25%    1426.250000    28.000000                        NaN
50%    2851.500000    43.000000                        NaN
75%    4276.750000    59.000000                        NaN
max    5702.000000   100.000000                        NaN

**Make Report Date TO DateTime**

In [14]:
casesday = cases['Report date']
casesnum = cases['Case no.']

In [25]:
cases['Date to report'] = pd.to_datetime(cases['Report date'], infer_datetime_format=True)
cases['Date to report'].head()

0   2020-01-23
1   2020-01-23
2   2020-01-24
3   2020-01-24
4   2020-01-24
Name: Date to report, dtype: datetime64[ns]

In [30]:
cases1 = cases.copy()

In [31]:
cases1.head()

Case no. Report date Date of onset Gender  Age  Name of hospital admitted  \
0         1  23/01/2020    21/01/2020      M   39                        NaN   
1         2  23/01/2020    18/01/2020      M   56                        NaN   
2         3  24/01/2020    20/01/2020      F   62                        NaN   
3         4  24/01/2020    23/01/2020      F   62                        NaN   
4         5  24/01/2020    23/01/2020      M   63                        NaN   

  Hospitalised/Discharged/Deceased HK/Non-HK resident Case classification*  \
0                       Discharged    Non-HK resident        Imported case   
1                       Discharged        HK resident        Imported case   
2                       Discharged    Non-HK resident        Imported case   
3                       Discharged    Non-HK resident        Imported case   
4                       Discharged    Non-HK resident        Imported case   

  Confirmed/probable Date to report  
0          Confirmed     2020-01-23  
1          Confirmed     2020-01-23  
2          Confirmed     2020-01-24  
3          Confirmed     2020-01-24  
4          Confirmed     2020-01-24

In [32]:
cases1.drop(['Report date', 'Name of hospital admitted'], axis = 1, inplace = True)
cases1.head()

Case no. Date of onset Gender  Age Hospitalised/Discharged/Deceased  \
0         1    21/01/2020      M   39                       Discharged   
1         2    18/01/2020      M   56                       Discharged   
2         3    20/01/2020      F   62                       Discharged   
3         4    23/01/2020      F   62                       Discharged   
4         5    23/01/2020      M   63                       Discharged   

  HK/Non-HK resident Case classification* Confirmed/probable Date to report  
0    Non-HK resident        Imported case          Confirmed     2020-01-23  
1        HK resident        Imported case          Confirmed     2020-01-23  
2    Non-HK resident        Imported case          Confirmed     2020-01-24  
3    Non-HK resident        Imported case          Confirmed     2020-01-24  
4    Non-HK resident        Imported case          Confirmed     2020-01-24

**Show all**

In [36]:
pd.set_option('max_columns',1000)
pd.set_option('max_row',5000)
pd.set_option('display.float_format', lambda x: '%.5f' % x)

In [37]:
cases1.loc[:,'Date of onset'].value_counts()

Asymptomatic    1599
27/07/2020       130
25/07/2020       118
24/07/2020       110
22/07/2020       104
21/07/2020       103
20/07/2020       101
23/07/2020        99
17/07/2020        96
26/07/2020        95
28/07/2020        93
16/07/2020        89
18/07/2020        88
15/07/2020        83
29/07/2020        80
14/07/2020        76
19/07/2020        71
01/08/2020        60
30/07/2020        59
31/07/2020        58
13/07/2020        57
03/08/2020        54
12/07/2020        52
02/08/2020        50
05/08/2020        50
Pending           47
04/08/2020        47
10/07/2020        45
19/03/2020        44
11/07/2020        43
24/03/2020        41
23/03/2020        41
16/03/2020        40
07/08/2020        40
07/07/2020        39
17/03/2020        39
18/03/2020        38
09/07/2020        38
20/03/2020        38
27/03/2020        37
25/03/2020        37
10/08/2020        36
08/08/2020        34
06/08/2020        32
09/08/2020        32
26/03/2020        32
22/03/2020        31
21/03/2020   

- **Asymptomatic 1599**
- **Pending 47**
- **Unknown 26**
- **January 1**
- **October 1**
- **Mid-July 1**
- **Mid-March 1**

- **For the convience of analysis, We can set the Asymptonmatic, Pending, Unknown's on setdate as the same as the reported date**

- **In the main table, set Asymptonmatic, Pending, Unknown to `NaN`**
- **And we alse need to take the Asymptomatic out, for a indi table to analysis**
- **January = 01/01/2020**
- **October = 01/10/2020**
- **Mid-July = 15/07/2020**
- **Mid-March = 15/03/2020**

In [74]:
# set Asymptonmatic, Pending, Unknown to NaN
casesForMain = cases1.copy()
casesForMain.loc[casesForMain['Date of onset']=='Asymptomatic', 'Date of onset'] = np.nan
casesForMain.loc[casesForMain['Date of onset']=='Pending', 'Date of onset'] = np.nan
casesForMain.loc[casesForMain['Date of onset']=='Unknown', 'Date of onset'] = np.nan

casesForMain.loc[casesForMain['Date of onset']=='January', 'Date of onset'] = '01/01/2020'
casesForMain.loc[casesForMain['Date of onset']=='October', 'Date of onset'] = '01/10/2020'
casesForMain.loc[casesForMain['Date of onset']=='Mid-July', 'Date of onset'] = '15/07/2020'
casesForMain.loc[casesForMain['Date of onset']=='Mid-March', 'Date of onset'] = '15/03/2020'

# print(casesForMain.loc[casesForMain['Date of onset']=='January', 'Date of onset'])
# print(casesForMain.loc[casesForMain['Date of onset']=='October', 'Date of onset'])
# print(casesForMain.loc[casesForMain['Date of onset']=='Mid-July', 'Date of onset'])
# print(casesForMain.loc[casesForMain['Date of onset']=='Mid-March', 'Date of onset'])

Series([], Name: Date of onset, dtype: object)
Series([], Name: Date of onset, dtype: object)
Series([], Name: Date of onset, dtype: object)
Series([], Name: Date of onset, dtype: object)


In [86]:
casesForMain['Onset Date'] = pd.to_datetime(casesForMain['Date of onset'], infer_datetime_format=True)
casesForMain.drop(['Date of onset'], axis=1, inplace=True)
casesForMain.head()

Case no. Gender  Age Hospitalised/Discharged/Deceased HK/Non-HK resident  \
0         1      M   39                       Discharged    Non-HK resident   
1         2      M   56                       Discharged        HK resident   
2         3      F   62                       Discharged    Non-HK resident   
3         4      F   62                       Discharged    Non-HK resident   
4         5      M   63                       Discharged    Non-HK resident   

  Case classification* Confirmed/probable Date to report Onset Date  
0        Imported case          Confirmed     2020-01-23 2020-01-21  
1        Imported case          Confirmed     2020-01-23 2020-01-18  
2        Imported case          Confirmed     2020-01-24 2020-01-20  
3        Imported case          Confirmed     2020-01-24 2020-01-23  
4        Imported case          Confirmed     2020-01-24 2020-01-23

# Hospitalised/Discharged/Deceased

In [88]:
casesForMain.loc[:,'Hospitalised/Discharged/Deceased'].value_counts()

Discharged        5267
Hospitalised       224
Deceased           108
To be provided      73
No admission        30
Name: Hospitalised/Discharged/Deceased, dtype: int64

In [91]:
# No admission 不适用
casesForMain[casesForMain['Case no.'] == 1913]

Case no. Gender  Age Hospitalised/Discharged/Deceased  \
1912      1913      M   27                     No admission   

     HK/Non-HK resident Case classification* Confirmed/probable  \
1912    Non-HK resident        Imported case          Confirmed   

     Date to report Onset Date  
1912     2020-07-20        NaT

In [94]:
# Hospitalised 住院
#casesForMain.loc[casesForMain['Hospitalised/Discharged/Deceased']=='Hospitalised']

In [95]:
#Deceased 死亡
#casesForMain.loc[casesForMain['Hospitalised/Discharged/Deceased']=='Deceased']

In [97]:
#To be provided 有待提供
# casesForMain.loc[casesForMain['Hospitalised/Discharged/Deceased']=='To be provided']

In [99]:
# set To be provided, No admission to nan
casesForMain.loc[casesForMain['Hospitalised/Discharged/Deceased']=='To be provided', 'Hospitalised/Discharged/Deceased'] = np.nan
casesForMain.loc[casesForMain['Hospitalised/Discharged/Deceased']=='No admission', 'Hospitalised/Discharged/Deceased'] = np.nan

# set Hospitalised = 2
casesForMain.loc[casesForMain['Hospitalised/Discharged/Deceased']=='Hospitalised', 'Hospitalised/Discharged/Deceased'] = 2
# set Deceased = 1, Discharged = 0
casesForMain.loc[casesForMain['Hospitalised/Discharged/Deceased']=='Deceased', 'Hospitalised/Discharged/Deceased'] = 1
casesForMain.loc[casesForMain['Hospitalised/Discharged/Deceased']=='Discharged', 'Hospitalised/Discharged/Deceased'] = 0

casesForMain.head()

Case no. Gender  Age Hospitalised/Discharged/Deceased HK/Non-HK resident  \
0         1      M   39                                0    Non-HK resident   
1         2      M   56                                0        HK resident   
2         3      F   62                                0    Non-HK resident   
3         4      F   62                                0    Non-HK resident   
4         5      M   63                                0    Non-HK resident   

  Case classification* Confirmed/probable Date to report Onset Date  
0        Imported case          Confirmed     2020-01-23 2020-01-21  
1        Imported case          Confirmed     2020-01-23 2020-01-18  
2        Imported case          Confirmed     2020-01-24 2020-01-20  
3        Imported case          Confirmed     2020-01-24 2020-01-23  
4        Imported case          Confirmed     2020-01-24 2020-01-23

## HK/Non-HK resident

In [100]:
casesForMain.loc[:,'HK/Non-HK resident'].value_counts()

HK resident        5419
Non-HK resident     231
HK Resident          42
Non-HK Resident       7
non-HK resident       3
Name: HK/Non-HK resident, dtype: int64

In [102]:
# set HK resident = 1
casesForMain.loc[casesForMain['HK/Non-HK resident']=='HK resident', 'HK/Non-HK resident'] = 1
casesForMain.loc[casesForMain['HK/Non-HK resident']=='HK Resident', 'HK/Non-HK resident'] = 1
# set Non Hk resident = 0
casesForMain.loc[casesForMain['HK/Non-HK resident']=='Non-HK resident', 'HK/Non-HK resident'] = 0
casesForMain.loc[casesForMain['HK/Non-HK resident']=='Non-HK Resident', 'HK/Non-HK resident'] = 0
casesForMain.loc[casesForMain['HK/Non-HK resident']=='non-HK resident', 'HK/Non-HK resident'] = 0

casesForMain['HK/Non-HK resident'].head()

0    0
1    1
2    0
3    0
4    0
Name: HK/Non-HK resident, dtype: object

## Case classification*

In [103]:
casesForMain.loc[:,'Case classification*'].value_counts()

Epidemiologically linked with local case             2470
Imported case                                        1586
Local case                                           1441
Possibly local case                                   103
Epidemiologically linked with possibly local case      62
Epidemiologically linked with imported case            40
Name: Case classification*, dtype: int64

In [108]:
casesForMain

Case no. Gender  Age Hospitalised/Discharged/Deceased  \
0            1      M   39                                0   
1            2      M   56                                0   
2            3      F   62                                0   
3            4      F   62                                0   
4            5      M   63                                0   
...        ...    ...  ...                              ...   
5697      5698      M   70                              NaN   
5698      5699      F   65                              NaN   
5699      5700      F   30                              NaN   
5700      5701      F   69                              NaN   
5701      5702      F   46                              NaN   

     HK/Non-HK resident                      Case classification*  \
0                     0                             Imported case   
1                     1                             Imported case   
2                     0                             Imported case   
3                     0                             Imported case   
4                     0                             Imported case   
...                 ...                                       ...   
5697                  1  Epidemiologically linked with local case   
5698                  1  Epidemiologically linked with local case   
5699                  1  Epidemiologically linked with local case   
5700                  1  Epidemiologically linked with local case   
5701                  1  Epidemiologically linked with local case   

     Confirmed/probable Date to report Onset Date  
0                     1     2020-01-23 2020-01-21  
1                     1     2020-01-23 2020-01-18  
2                     1     2020-01-24 2020-01-20  
3                     1     2020-01-24 2020-01-23  
4                     1     2020-01-24 2020-01-23  
...                 ...            ...        ...  
5697                  1     2020-11-23        NaT  
5698                  1     2020-11-23        NaT  
5699                  1     2020-11-23 2020-11-18  
5700                  1     2020-11-23        NaT  
5701                  1     2020-11-23 2020-11-22  

[5702 rows x 9 columns]

## Confirmed/probable	

In [109]:
casesForMain.loc[:,'Confirmed/probable'].value_counts()

1    5701
0       1
Name: Confirmed/probable, dtype: int64

In [107]:
casesForMain.loc[casesForMain['Confirmed/probable']=='Confirmed', 'Confirmed/probable'] = 1
casesForMain.loc[casesForMain['Confirmed/probable']=='Probable', 'Confirmed/probable'] = 0

casesForMain.head()

Case no. Gender  Age Hospitalised/Discharged/Deceased HK/Non-HK resident  \
0         1      M   39                                0                  0   
1         2      M   56                                0                  1   
2         3      F   62                                0                  0   
3         4      F   62                                0                  0   
4         5      M   63                                0                  0   

  Case classification* Confirmed/probable Date to report Onset Date  
0        Imported case                  1     2020-01-23 2020-01-21  
1        Imported case                  1     2020-01-23 2020-01-18  
2        Imported case                  1     2020-01-24 2020-01-20  
3        Imported case                  1     2020-01-24 2020-01-23  
4        Imported case                  1     2020-01-24 2020-01-23

## Make a indi table for Asymptomatic

In [87]:
casesForAsy = cases1.loc[cases1['Date of onset']=='Asymptomatic'].copy()
casesForAsy.drop(['Date of onset'], axis = 1, inplace = True)
casesForAsy.head()

Case no. Gender  Age Hospitalised/Discharged/Deceased HK/Non-HK resident  \
79        80      M   57                       Discharged        HK resident   
80        81      F   57                       Discharged        HK resident   
83        84      M   24                       Discharged        HK resident   
93        94      F   61                       Discharged        HK resident   
96        97      M   56                       Discharged        HK resident   

                                 Case classification* Confirmed/probable  \
79                                      Imported case          Confirmed   
80                                      Imported case          Confirmed   
83  Epidemiologically linked with possibly local case          Confirmed   
93                                      Imported case          Confirmed   
96                                      Imported case           Probable   

   Date to report  
79     2020-02-24  
80     2020-02-24  
83     2020-02-25  
93     2020-02-28  
96     2020-03-01